packages/editor/src/components/post-author/enhanced-post-author.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import { debounce } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import ComboboxControl from '../../../../components/build/combobox-control/';
+
+function EnhancedPostAuthor( { postAuthor, id, authors, onUpdateAuthor } ) {
+	const currentPostAuthor = postAuthor[ 0 ];
+	let initialAuthors = [];
+	authors.forEach( ( author, i ) => {
+		initialAuthors = [
+			...initialAuthors,
+			{
+				key: i,
+				name: author.name,
+				value: author.name,
+				id: author.id,
+			},
+		];
+	} );
+
+	// Ensure the current author is included in the initial dropdown list.
+	let foundAuthor = initialAuthors.findIndex(
+		( author ) => currentPostAuthor && currentPostAuthor.id === author.id
+	);
+
+	if ( currentPostAuthor && foundAuthor < 0 ) {
+		currentPostAuthor.key = initialAuthors.length;
+		initialAuthors = [ currentPostAuthor, ...initialAuthors ];
+		foundAuthor = 0;
+	}
+
+	// The search result loading state.
+	const [ isLoadingSearchResults, setIsLoadingSearchResult ] = useState(
+		false
+	);
+
+	// The list of authors available in the dropdown.
+	const [ availableAuthors, setAvailableAuthors ] = useState(
+		initialAuthors
+	);
+
+	// The currently selected author.
+	const [ selectedAuthor, setSelectedAuthor ] = useState( currentPostAuthor );
```

Comment:
There's no need to mirror the controlled props in local state like this, `currentPostAuthor` updates.

Reply:
this is used to track the current selection so we know not to search when the typed (input) value matches the current user, may no longer be needed - i'll test removing.

Reply:
without this, once you select an author the search becomes active and the dropdown shows only your user. with this, the dropdown goes back to showing the default users. I will remove for now to simplify and we can consider the best user interaction.